# ADR013 - Valid ADR Links

Links to other ADR documents should point to existing files.

## Why This Rule Exists

Broken ADR links:
- Prevent readers from following decision history
- Indicate missing or deleted ADRs
- Create confusion in documentation

## Format

Links to other ADRs should reference existing files:

```markdown
## Status

Superseded by [ADR-0005](0005-use-kubernetes.md)

## Context

This builds on [ADR-0002](0002-container-strategy.md).
```

## Examples

### Incorrect

```markdown
See [ADR-0099](0099-nonexistent.md) for details.
```

File `0099-nonexistent.md` doesn't exist.

### Correct

```markdown
See [ADR-0002](0002-use-rust.md) for context.
```

File `0002-use-rust.md` exists.

## Collection Rule

This rule analyzes all ADR documents together to validate cross-references.

## What's Checked

- Links with `.md` extension in ADR directories
- Relative paths are resolved from the source document
- Both filename-only and path references

## Rule Details

- **Rule ID**: ADR013
- **Name**: adr-valid-adr-links
- **Category**: Links
- **Severity**: Warning
- **Type**: Collection rule (multi-document)
- **Automatic Fix**: Not available

## Related Rules

- [ADR010](./adr010.md) - Superseded ADRs reference replacement
- [ADR011](./adr011.md) - Sequential numbering (helps identify missing ADRs)
