# ADR005 - Required Decision Section

ADRs must have a decision section stating what was decided.

## Why This Rule Exists

The decision section:
- Clearly states what was chosen
- Makes the outcome unambiguous
- Provides the actionable result

## Formats

### Nygard Format

A `## Decision` section:

```markdown
## Decision

We will use Rust for the new microservice implementation.
```

### MADR Format

A `## Decision Outcome` section:

```markdown
## Decision Outcome

Chosen option: "Rust", because it provides memory safety without garbage collection
and has excellent performance characteristics.
```

## Examples

### Incorrect

```markdown
# 1. Use Rust

Date: 2024-01-15

## Status

Accepted

## Context

We need a language.

## Consequences

Team training needed.
```

Missing Decision section.

### Correct (Nygard)

```markdown
## Decision

We will use Rust for the new microservice implementation.
```

### Correct (MADR)

```markdown
## Decision Outcome

Chosen option: "Rust", because it provides memory safety and performance.
```

## Rule Details

- **Rule ID**: ADR005
- **Name**: adr-required-decision
- **Category**: Structure
- **Severity**: Error
- **Automatic Fix**: Not available

## Related Rules

- [ADR014](./adr014.md) - Decision should have meaningful content
