# ADR008 - Date Format

ADR dates must follow ISO 8601 format (YYYY-MM-DD).

## Why This Rule Exists

ISO 8601 format:
- Is unambiguous internationally
- Sorts correctly alphabetically
- Is machine-readable
- Follows industry standards

## Format

Dates must be in `YYYY-MM-DD` format:

```markdown
Date: 2024-01-15
```

Or in MADR frontmatter:

```yaml
---
date: 2024-01-15
---
```

## Examples

### Incorrect

```markdown
Date: January 15, 2024
Date: 15/01/2024
Date: 01-15-2024
Date: 2024/01/15
```

### Correct

```markdown
Date: 2024-01-15
Date: 2024-12-31
```

```yaml
---
date: 2024-01-15
---
```

## Rule Details

- **Rule ID**: ADR008
- **Name**: adr-date-format
- **Category**: Validation
- **Severity**: Error
- **Automatic Fix**: Not available

## Related Rules

- [ADR003](./adr003.md) - Date is required
