# ADR001 - Title Format

ADR titles should follow the appropriate format for the document type.

## Why This Rule Exists

Consistent title formatting makes ADRs:
- Easy to identify and reference
- Sortable by number
- Recognizable as ADRs in search results

## Formats

### Nygard Format

Titles must follow the pattern `# N. Title` where N is the ADR number:

```markdown
# 1. Record architecture decisions
# 42. Use PostgreSQL for persistence
```

### MADR Format

Titles should be a simple H1 heading (no number required):

```markdown
# Use PostgreSQL for persistence
```

## Examples

### Incorrect (Nygard)

```markdown
# Use Rust for implementation
```

Missing number prefix.

### Correct (Nygard)

```markdown
# 1. Use Rust for implementation
```

### Incorrect (MADR)

Missing H1 title entirely or using wrong heading level.

### Correct (MADR)

```markdown
---
status: accepted
date: 2024-01-15
---

# Use Rust for implementation
```

## Rule Details

- **Rule ID**: ADR001
- **Name**: adr-title-format
- **Category**: Structure
- **Severity**: Error
- **Automatic Fix**: Not available

## Configuration

This rule has no configuration options. Format is auto-detected based on document content.

## Related Rules

- [ADR009](./adr009.md) - Filename should match ADR number
