# ADR007 - Valid Status Value

ADR status must be a recognized value.

## Why This Rule Exists

Standard status values ensure:
- Consistent interpretation across teams
- Clear lifecycle management
- Tooling compatibility

## Valid Status Values

Default valid statuses (case-insensitive):
- `proposed` - Under consideration
- `accepted` - Approved and active
- `deprecated` - No longer recommended
- `superseded` - Replaced by another ADR
- `rejected` - Considered but not adopted

## Examples

### Incorrect

```markdown
## Status

Maybe

## Status

In Progress
```

### Correct

```markdown
## Status

Accepted

## Status

Proposed

## Status

Superseded by ADR-0005
```

## Configuration

Customize valid statuses:

```toml
[ADR007]
valid-statuses = ["proposed", "accepted", "deprecated", "superseded", "rejected", "draft"]
```

## Rule Details

- **Rule ID**: ADR007
- **Name**: adr-valid-status
- **Category**: Validation
- **Severity**: Error
- **Automatic Fix**: Not available

## Related Rules

- [ADR002](./adr002.md) - Status is required
- [ADR010](./adr010.md) - Superseded ADRs should reference replacement
