# ADR012 - No Duplicate Numbers

Each ADR number must be unique across all ADRs.

## Why This Rule Exists

Duplicate numbers cause:
- Confusion about which ADR to reference
- Broken cross-references
- Difficulty navigating ADR history

## Examples

### Incorrect

```
0001-record-decisions.md      # ADR 1
0001-use-rust.md              # ADR 1 (duplicate!)
0002-database-choice.md       # ADR 2
```

### Correct

```
0001-record-decisions.md      # ADR 1
0002-use-rust.md              # ADR 2
0003-database-choice.md       # ADR 3
```

## Collection Rule

This rule analyzes all ADR documents in a directory together to detect duplicates.

## Rule Details

- **Rule ID**: ADR012
- **Name**: adr-no-duplicate-numbers
- **Category**: Structure
- **Severity**: Error
- **Type**: Collection rule (multi-document)
- **Automatic Fix**: Not available

## Related Rules

- [ADR009](./adr009.md) - Filename matches title number
- [ADR011](./adr011.md) - Sequential numbering
