# ADR011 - Sequential Numbering

ADR numbers should be sequential with no gaps.

## Why This Rule Exists

Sequential numbering:
- Makes it easy to identify missing ADRs
- Provides clear ordering
- Indicates the decision timeline
- Helps with navigation

## Format

ADRs should be numbered starting from 1 (or 0) without gaps:

```
0001-record-decisions.md      # ADR 1
0002-use-rust.md              # ADR 2
0003-database-choice.md       # ADR 3
```

## Examples

### Incorrect

```
0001-record-decisions.md      # ADR 1
0003-database-choice.md       # ADR 3 (gap - where is 2?)
0004-use-kubernetes.md        # ADR 4
```

### Correct

```
0001-record-decisions.md      # ADR 1
0002-use-rust.md              # ADR 2
0003-database-choice.md       # ADR 3
0004-use-kubernetes.md        # ADR 4
```

## Collection Rule

This rule analyzes all ADR documents in a directory together to check for gaps.

## Note on Rejected ADRs

Even rejected ADRs should keep their numbers. Don't delete or renumber ADRs:

```
0001-record-decisions.md      # Accepted
0002-use-java.md              # Rejected (keep it!)
0003-use-rust.md              # Accepted (supersedes thinking in 0002)
```

## Rule Details

- **Rule ID**: ADR011
- **Name**: adr-sequential-numbering
- **Category**: Structure
- **Severity**: Warning
- **Type**: Collection rule (multi-document)
- **Automatic Fix**: Not available

## Related Rules

- [ADR009](./adr009.md) - Filename matches title number
- [ADR012](./adr012.md) - No duplicate numbers
