# ADR016 - Considered Options Format

In MADR format, the Considered Options section should list at least 2 options.

## Why This Rule Exists

Multiple options demonstrate:
- Alternatives were evaluated
- Due diligence was performed
- The decision wasn't predetermined
- Trade-offs were considered

## Format

List at least 2 options as bullets or H3 headings:

```markdown
## Considered Options

* PostgreSQL
* MySQL
* MongoDB
```

Or with detailed subsections:

```markdown
## Considered Options

### PostgreSQL

A mature relational database with excellent SQL support.

### MySQL

A widely-used relational database with good performance.

### MongoDB

A document database for flexible schemas.
```

## Examples

### Incorrect

```markdown
## Considered Options

* PostgreSQL
```

Only one option listed.

### Correct

```markdown
## Considered Options

* PostgreSQL
* MySQL
* MongoDB
```

## Configuration

Customize minimum options:

```toml
[ADR016]
min-options = 2  # Default
```

## Rule Details

- **Rule ID**: ADR016
- **Name**: adr-considered-options-format
- **Category**: Content
- **Severity**: Info
- **Applies to**: MADR format only
- **Automatic Fix**: Not available

## Related Rules

- [ADR015](./adr015.md) - Decision Drivers format
- [ADR017](./adr017.md) - Consequences structure
