# ADR010 - Superseded ADRs Reference Replacement

ADRs with "Superseded" status should reference the ADR that replaces them.

## Why This Rule Exists

When an ADR is superseded:
- Readers need to know which ADR to follow instead
- The decision history remains traceable
- Teams can understand the evolution of decisions

## Format

The status section should include a link to the replacement ADR:

```markdown
## Status

Superseded by [ADR-0005](0005-use-kubernetes.md)
```

Or in MADR:

```markdown
---
status: superseded
superseded-by: ADR-0005
---
```

## Examples

### Incorrect

```markdown
## Status

Superseded
```

No reference to replacement ADR.

### Correct

```markdown
## Status

Superseded by [ADR-0005](0005-use-kubernetes.md)
```

```markdown
## Status

Superseded by [0005-use-kubernetes.md](./0005-use-kubernetes.md)
```

## Collection Rule

This rule analyzes multiple ADR documents together to validate cross-references.

## Rule Details

- **Rule ID**: ADR010
- **Name**: adr-superseded-has-replacement
- **Category**: Links
- **Severity**: Warning
- **Type**: Collection rule (multi-document)
- **Automatic Fix**: Not available

## Related Rules

- [ADR007](./adr007.md) - Valid status values
- [ADR013](./adr013.md) - Valid ADR links
