# ADR014 - Non-Empty Sections

Required ADR sections should have meaningful content, not placeholders.

## Why This Rule Exists

Placeholder content:
- Provides no value to readers
- Indicates incomplete documentation
- May mislead about decision completeness

## Placeholder Patterns Detected

- Empty sections
- `TODO`, `TBD`, `To be determined`
- `...` (ellipsis)
- `[Insert here]`, `<Insert here>`
- `Lorem ipsum`
- `Fill in`, `Add content`, `Write here`
- Very short content (< 3 characters)

## Examples

### Incorrect

```markdown
## Context

TODO: Fill in context later

## Decision

TBD

## Consequences

...
```

### Correct

```markdown
## Context

We need to choose a programming language for our new microservice.
The team has experience with Java, Python, and Rust. Performance
and memory safety are key requirements.

## Decision

We will use Rust for the new microservice implementation.

## Consequences

Team members will need Rust training, which will take 2-3 weeks.
Build times will be longer initially but runtime performance will improve.
```

## Rule Details

- **Rule ID**: ADR014
- **Name**: adr-non-empty-sections
- **Category**: Content
- **Severity**: Warning
- **Automatic Fix**: Not available

## Related Rules

- [ADR004](./adr004.md) - Required context section
- [ADR005](./adr005.md) - Required decision section
- [ADR006](./adr006.md) - Required consequences section
