# ADR015 - Decision Drivers Format

In MADR format, the Decision Drivers section should use a bullet list.

## Why This Rule Exists

Bullet lists for decision drivers:
- Clearly enumerate factors
- Make drivers easy to scan
- Enable consistent formatting
- Help readers understand decision criteria

## Format

Decision Drivers should be a bullet list:

```markdown
## Decision Drivers

* Need ACID compliance for data integrity
* Team familiarity with SQL databases
* Strong ecosystem and tooling support
* Active community and long-term viability
```

## Examples

### Incorrect

```markdown
## Decision Drivers

We need ACID compliance and team familiarity with SQL.
Also good tooling is important.
```

Paragraph text instead of bullet list.

### Correct

```markdown
## Decision Drivers

* Need ACID compliance for data integrity
* Team familiarity with SQL databases
* Strong ecosystem and tooling support
```

Or with dashes:

```markdown
## Decision Drivers

- Need ACID compliance
- Team familiarity
- Good tooling
```

## Rule Details

- **Rule ID**: ADR015
- **Name**: adr-decision-drivers-format
- **Category**: Structure
- **Severity**: Info
- **Applies to**: MADR format only
- **Automatic Fix**: Not available

## Related Rules

- [ADR016](./adr016.md) - Considered Options format
