# ADR003 - Required Date

ADRs must include the date when the decision was made.

## Why This Rule Exists

Dates provide:
- Historical context for when decisions were made
- Timeline for understanding decision evolution
- Reference for when to revisit decisions

## Formats

### Nygard Format

Date appears on a line after the title:

```markdown
# 1. Use Rust

Date: 2024-01-15
```

### MADR Format

Date is in YAML frontmatter:

```markdown
---
status: accepted
date: 2024-01-15
---
```

## Examples

### Incorrect

```markdown
# 1. Use Rust

## Status

Accepted
```

Missing date.

### Correct (Nygard)

```markdown
# 1. Use Rust

Date: 2024-01-15

## Status

Accepted
```

### Correct (MADR)

```markdown
---
status: accepted
date: 2024-01-15
---

# Use Rust
```

## Rule Details

- **Rule ID**: ADR003
- **Name**: adr-required-date
- **Category**: Structure
- **Severity**: Error
- **Automatic Fix**: Not available

## Related Rules

- [ADR008](./adr008.md) - Date format validation
