# MD004 - Unordered List Style

Unordered list style should be consistent.

## Why This Rule Exists

Markdown supports three markers for unordered lists: `-`, `*`, and `+`. Using
different markers inconsistently creates visual noise and can indicate
accidental mixing of content from different sources.

## Examples

### Incorrect

```markdown
- Item one
* Item two
+ Item three
```

### Correct

```markdown
- Item one
- Item two
- Item three
```

Or consistently using asterisks:

```markdown
* Item one
* Item two
* Item three
```

## Configuration

```toml
[MD004]
style = "dash"  # Options: "dash", "asterisk", "plus", "consistent"
```

| Value | Marker | Example |
|-------|--------|---------|
| `dash` | `-` | `- Item` |
| `asterisk` | `*` | `* Item` |
| `plus` | `+` | `+ Item` |
| `consistent` | First used | Matches first list marker |

## When to Disable

- Documents intentionally using different markers to distinguish list types
- Importing content from multiple sources

## Rule Details

- **Rule ID**: MD004
- **Aliases**: ul-style
- **Category**: Formatting
- **Severity**: Warning
- **Auto-fix**: Yes

## Related Rules

- [MD005](./md005.md) - List item indentation
- [MD006](./md006.md) - Lists start at beginning of line
- [MD007](./md007.md) - Unordered list indentation
- [MD029](./md029.md) - Ordered list prefix style
