# MD005 - List Item Indentation

List item indentation should be consistent within a list.

## Why This Rule Exists

Inconsistent indentation within lists can cause rendering issues and makes
documents harder to read in source form. Proper indentation also ensures
nested lists render correctly.

## Examples

### Incorrect

```markdown
- Item one
 - Item two (wrong indentation)
- Item three
```

### Correct

```markdown
- Item one
- Item two
- Item three
```

### Nested Lists (Correct)

```markdown
- Item one
  - Nested item
  - Another nested
- Item two
```

## Configuration

This rule has no configuration options. It enforces consistent indentation
within each list.

## When to Disable

- Working with auto-generated content that has intentional spacing
- Documents with complex nested structures requiring manual control

## Rule Details

- **Rule ID**: MD005
- **Aliases**: list-indent
- **Category**: Formatting
- **Severity**: Warning
- **Auto-fix**: Yes

## Related Rules

- [MD004](./md004.md) - Unordered list style
- [MD006](./md006.md) - Lists start at beginning of line
- [MD007](./md007.md) - Unordered list indentation depth
- [MD030](./md030.md) - Spaces after list markers
