# MD001 - Heading Increment

{{#rustdoc_include ../../../../crates/mdbook-lint-rulesets/src/standard/md001.rs:9:57}}

## Rule Details

- **Rule ID**: MD001
- **Aliases**: heading-increment
- **Category**: Structure
- **Severity**: Warning
- **Automatic Fix**: Not available

## Common Violations and Solutions

### Skipping from H1 to H3

**Problem:**
```markdown
# Main Title
### Subsection
```

**Solution:**
```markdown
# Main Title
## Section
### Subsection
```

### Deep Nesting Without Intermediate Levels

**Problem:**
```markdown
## Chapter
##### Deep Detail
```

**Solution:**
```markdown
## Chapter
### Section
#### Subsection
##### Deep Detail
```

## Rationale for Sequential Headings

1. **Accessibility**: Screen readers rely on proper heading hierarchy to help users navigate documents
2. **Document Structure**: Sequential headings create a logical outline
3. **SEO**: Search engines use heading structure to understand content hierarchy
4. **Table of Contents**: Automated TOC generators expect proper nesting

## Integration with mdBook

mdBook's sidebar generation relies on proper heading structure. Violating this rule can lead to:
- Broken navigation in the sidebar
- Incorrect TOC generation
- Poor mobile navigation experience

## Related Rules

- [MD002](./md002.md) - First heading should be a top-level heading
- [MD003](./md003.md) - Heading style consistency
- [MD022](./md022.md) - Headings should be surrounded by blank lines
- [MD025](./md025.md) - Multiple top-level headings in the same document

## References

- [WCAG 2.1 - Heading Levels](https://www.w3.org/WAI/WCAG21/Techniques/general/G141)
- [MDN - Heading Elements](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Heading_Elements)
- [CommonMark Spec - ATX Headings](https://spec.commonmark.org/0.30/#atx-headings)