# MD006 - List Start Left

Consider starting bulleted lists at the beginning of the line.

## Why This Rule Exists

Lists that don't start at the beginning of the line can cause unexpected
rendering behavior in some Markdown parsers. Starting lists at column 0
ensures consistent rendering across all platforms.

## Examples

### Incorrect

```markdown
Some text:
  - Indented list item
  - Another indented item
```

### Correct

```markdown
Some text:

- List item at start of line
- Another item at start of line
```

### Nested Lists (Allowed)

```markdown
- Top level item
  - Nested item (indentation is fine for nesting)
  - Another nested item
```

## Configuration

This rule has no configuration options.

## When to Disable

- Documents with intentionally indented lists for specific formatting
- Content that uses indentation for semantic meaning

## Rule Details

- **Rule ID**: MD006
- **Aliases**: ul-start-left
- **Category**: Formatting
- **Severity**: Warning
- **Auto-fix**: Yes

## Related Rules

- [MD004](./md004.md) - Unordered list style
- [MD005](./md005.md) - List item indentation consistency
- [MD007](./md007.md) - Unordered list indentation
- [MD023](./md023.md) - Headings start at beginning of line
