# MD019 - Multiple Spaces After Hash on ATX Style Heading

**Severity**: Warning  
**Category**: Headings  
**Auto-fix**: ✓ Available

## Rule Description

This rule ensures there's only a single space after the hash character(s) in ATX-style headings. Multiple spaces are unnecessary and can cause inconsistent formatting.

## Why This Rule Exists

Single space after hash is important because:
- Maintains consistent formatting across documents
- Follows standard markdown conventions
- Reduces unnecessary whitespace
- Improves readability and predictability

## Examples

### ❌ Incorrect (violates rule)

```markdown
#  Heading with multiple spaces
##   Another heading with extra spaces
###    Too many spaces here
```

### ✅ Correct

```markdown
# Heading with single space
## Another heading correctly formatted
### Proper spacing
```

## Configuration

This rule has no configuration options.

## Automatic Fix

This rule supports automatic fixing with `--fix`. The fix will:
- Reduce multiple spaces to a single space after hash character(s)
- Preserve the heading level and content
- Handle all heading levels (1-6)

### Apply Fix

```bash
# Fix multiple space issues in headings
mdbook-lint lint --fix docs/

# Preview what would be fixed
mdbook-lint lint --fix --dry-run docs/
```

## When to Disable

Consider disabling this rule if:
- Your style guide requires multiple spaces for alignment
- You're maintaining legacy content with specific spacing requirements

### Disable in Config

```toml
# .mdbook-lint.toml
disabled_rules = ["MD019"]
```

### Disable Inline

```markdown
<!-- mdbook-lint-disable MD019 -->
##   Heading with multiple spaces allowed
<!-- mdbook-lint-enable MD019 -->
```

## Related Rules

- [MD018](./md018.html) - No space after hash on ATX heading
- [MD020](./md020.html) - No space inside hashes on closed ATX heading
- [MD021](./md021.html) - Multiple spaces inside hashes on closed ATX heading
- [MD022](./md022.html) - Headings should be surrounded by blank lines
- [MD023](./md023.html) - Headings must start at the beginning of the line

## References

- [Original markdownlint rule](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md019)
- [CommonMark Specification - ATX Headings](https://spec.commonmark.org/0.30/#atx-headings)