# MD012 - Multiple Consecutive Blank Lines

**Severity**: Warning  
**Category**: Whitespace  
**Auto-fix**: ✓ Available

## Rule Description

This rule checks for multiple consecutive blank lines in the document. Excessive blank lines add unnecessary whitespace without improving readability.

## Why This Rule Exists

Multiple consecutive blank lines create issues:
- Inconsistent spacing throughout documents
- Unnecessary vertical space in rendered output
- Potential confusion about section boundaries
- Increased file size without benefit

## Examples

### ❌ Incorrect (violates rule)

```markdown
# Heading

First paragraph.


Second paragraph with too many blank lines above.



Third paragraph with even more blank lines.
```

### ✅ Correct

```markdown
# Heading

First paragraph.

Second paragraph with single blank line.

Third paragraph properly spaced.
```

## Configuration

```toml
[rules.MD012]
maximum = 1  # Maximum consecutive blank lines allowed (default: 1)
```

## Automatic Fix

This rule supports automatic fixing with `--fix`. The fix will:
- Reduce multiple consecutive blank lines to the configured maximum
- Preserve intentional spacing at the maximum level
- Handle blank lines anywhere in the document

### Apply Fix

```bash
# Fix excessive blank lines
mdbook-lint lint --fix docs/

# Preview what would be fixed
mdbook-lint lint --fix --dry-run docs/
```

## When to Disable

Consider disabling this rule if:
- Your style guide requires multiple blank lines for visual separation
- You're working with generated content that uses specific spacing
- You need extra spacing for ASCII art or diagrams

### Disable in Config

```toml
# .mdbook-lint.toml
disabled_rules = ["MD012"]
```

### Disable Inline

```markdown
<!-- mdbook-lint-disable MD012 -->
Content with multiple


blank lines allowed here
<!-- mdbook-lint-enable MD012 -->
```

## Related Rules

- [MD009](./md009.html) - No trailing spaces
- [MD010](./md010.html) - Hard tabs
- [MD047](./md047.html) - Files should end with newline

## References

- [Original markdownlint rule](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md012)