# MD011 - Reversed Link Syntax

Reversed link syntax should be corrected.

## Why This Rule Exists

A common typo when writing Markdown links is reversing the bracket order,
writing `](url)[text` instead of `[text](url)`. This rule catches these
mistakes before they break your rendered documentation.

## Examples

### Incorrect

```markdown
Check out ](https://example.com)[this link for more info.

See ](./other-page.md)[the other page.
```

### Correct

```markdown
Check out [this link](https://example.com) for more info.

See [the other page](./other-page.md).
```

## Configuration

This rule has no configuration options.

## When to Disable

- Generally should not be disabled as reversed links never render correctly

## Rule Details

- **Rule ID**: MD011
- **Aliases**: no-reversed-links
- **Category**: Content
- **Severity**: Error
- **Auto-fix**: Yes (swaps to correct order)

## Related Rules

- [MD034](./md034.md) - Bare URLs
- [MD039](./md039.md) - Spaces inside link text
- [MD042](./md042.md) - Empty links
