# MD010 - Hard Tabs

**Severity**: Warning  
**Category**: Whitespace  
**Auto-fix**: ✓ Available

## Rule Description

This rule checks for hard tab characters in the document and suggests replacing them with spaces for consistency.

## Why This Rule Exists

Hard tabs can cause formatting inconsistencies:
- Tab width varies between editors (2, 4, or 8 spaces)
- Mixing tabs and spaces leads to misaligned text
- Different markdown renderers may handle tabs differently
- Code blocks and indentation become unpredictable

## Examples

### ❌ Incorrect (violates rule)

```text
→   This line starts with a tab
-→  List item with tab after marker
```→   Code block with tab indent
```

(Where → represents a tab character)

### ✅ Correct

```markdown
    This line uses spaces for indentation
-   List item with spaces after marker
```    Code block with space indent
```

## Configuration

```toml
[rules.MD010]
code_blocks = true  # Check for tabs in code blocks (default: true)
spaces_per_tab = 4  # Number of spaces to replace each tab with (default: 4)
```

## Automatic Fix

This rule supports automatic fixing with `--fix`. The fix will:
- Replace each tab character with the configured number of spaces
- Preserve the visual indentation of your content
- Handle tabs in all contexts (text, lists, code blocks)

### Apply Fix

```bash
# Fix all tab issues in your markdown files
mdbook-lint lint --fix docs/

# Preview what would be fixed
mdbook-lint lint --fix --dry-run docs/
```

## When to Disable

Consider disabling this rule if:
- Your project requires hard tabs (e.g., Makefiles in code examples)
- You're working with legacy content that uses tabs consistently
- Your team has standardized on tabs instead of spaces

### Disable in Config

```toml
# .mdbook-lint.toml
disabled_rules = ["MD010"]
```

### Disable Inline

```markdown
<!-- mdbook-lint-disable MD010 -->
→   Content with tabs allowed here
<!-- mdbook-lint-enable MD010 -->
```

## Related Rules

- [MD009](./md009.html) - No trailing spaces
- [MD012](./md012.html) - Multiple consecutive blank lines
- [MD047](./md047.html) - Files should end with newline

## References

- [Original markdownlint rule](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md010)
- [Tabs vs Spaces debate](https://www.jwz.org/doc/tabs-vs-spaces.html)