# MD027 - Multiple Spaces After Blockquote Symbol

**Severity**: Warning  
**Category**: Blockquotes  
**Auto-fix**: ✓ Available

## Rule Description

This rule ensures there's only a single space (or no space) after the blockquote marker (`>`). Multiple spaces create inconsistent formatting.

## Why This Rule Exists

Consistent blockquote spacing is important because:
- Maintains uniform appearance across documents
- Reduces unnecessary whitespace
- Follows standard markdown conventions
- Improves readability and predictability

## Examples

### ❌ Incorrect (violates rule)

```markdown
>  Multiple spaces after blockquote marker
>   Even more spaces here
>    Too much spacing
```

### ✅ Correct

```markdown
> Single space after marker
> Consistent spacing throughout
> Clean and readable

>No space is also valid
>When configured appropriately
```

## Configuration

```toml
[rules.MD027]
spaces = 1  # Number of spaces after blockquote marker (default: 1, can be 0)
```

## Automatic Fix

This rule supports automatic fixing with `--fix`. The fix will:
- Adjust spaces after blockquote markers to match configuration
- Preserve blockquote content and nesting
- Handle multi-level blockquotes correctly

### Apply Fix

```bash
# Fix blockquote spacing
mdbook-lint lint --fix docs/

# Preview what would be fixed
mdbook-lint lint --fix --dry-run docs/
```

## When to Disable

Consider disabling this rule if:
- Your style guide has different blockquote spacing requirements
- You're preserving legacy content with specific formatting
- You need variable spacing for visual emphasis

### Disable in Config

```toml
# .mdbook-lint.toml
disabled_rules = ["MD027"]
```

### Disable Inline

```markdown
<!-- mdbook-lint-disable MD027 -->
>   Blockquote with custom spacing
<!-- mdbook-lint-enable MD027 -->
```

## Related Rules

- [MD028](./md028.html) - Blank line inside blockquote
- [MD032](./md032.html) - Lists surrounded by blank lines

## References

- [Original markdownlint rule](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md027)
- [CommonMark Specification - Block Quotes](https://spec.commonmark.org/0.30/#block-quotes)