# MD021 - Multiple Spaces Inside Hashes on Closed ATX Style Heading

**Severity**: Warning  
**Category**: Headings  
**Auto-fix**: ✓ Available

## Rule Description

This rule ensures closed ATX-style headings have only a single space inside the hash markers. Multiple spaces create inconsistent formatting and unnecessary whitespace.

## Why This Rule Exists

Single space inside closed headings is important because:
- Maintains consistent formatting
- Follows standard markdown conventions
- Improves readability
- Ensures proper rendering across different parsers

## Examples

### ❌ Incorrect (violates rule)

```markdown
#  Heading with multiple spaces  #
##   Another heading   ##
###    Too many internal spaces    ###
```

### ✅ Correct

```markdown
# Heading with single spaces #
## Another heading ##
### Properly spaced heading ###
# Open heading is also fine
```

## Configuration

This rule has no configuration options.

## Automatic Fix

This rule supports automatic fixing with `--fix`. The fix will:
- Reduce multiple spaces to single spaces inside hash markers
- Preserve the heading level and content
- Maintain the closed heading style

### Apply Fix

```bash
# Fix multiple spaces in closed headings
mdbook-lint lint --fix docs/

# Preview what would be fixed
mdbook-lint lint --fix --dry-run docs/
```

## When to Disable

Consider disabling this rule if:
- Your style guide requires multiple spaces for alignment
- You prefer open ATX headings (without closing hashes)

### Disable in Config

```toml
# .mdbook-lint.toml
disabled_rules = ["MD021"]
```

### Disable Inline

```markdown
<!-- mdbook-lint-disable MD021 -->
##   Heading with multiple spaces   ##
<!-- mdbook-lint-enable MD021 -->
```

## Related Rules

- [MD018](./md018.html) - No space after hash on ATX heading
- [MD019](./md019.html) - Multiple spaces after hash on ATX heading
- [MD020](./md020.html) - No space inside hashes on closed ATX heading
- [MD022](./md022.html) - Headings should be surrounded by blank lines
- [MD003](./md003.html) - Heading style

## References

- [Original markdownlint rule](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md021)
- [CommonMark Specification - ATX Headings](https://spec.commonmark.org/0.30/#atx-headings)