# MD022 - Blanks Around Headings

Headings should be surrounded by blank lines.

## Why This Rule Exists

Blank lines around headings improve readability and ensure consistent rendering
across Markdown parsers. Some parsers require blank lines to properly recognize
headings.

## Examples

### Incorrect

```markdown
Some paragraph text.
## Heading
More text here.
```

### Correct

```markdown
Some paragraph text.

## Heading

More text here.
```

## Configuration

```toml
[MD022]
lines_above = 1  # Blank lines before heading (default: 1)
lines_below = 1  # Blank lines after heading (default: 1)
```

## When to Disable

- Documents with compact formatting requirements
- Content where headings immediately follow other headings

## Rule Details

- **Rule ID**: MD022
- **Aliases**: blanks-around-headings
- **Category**: Structure
- **Severity**: Warning
- **Auto-fix**: Yes

## Related Rules

- [MD023](./md023.md) - Headings start at beginning of line
- [MD031](./md031.md) - Blanks around fenced code blocks
- [MD032](./md032.md) - Blanks around lists
