# MD020 - No Space Inside Hashes on Closed ATX Style Heading

**Severity**: Warning  
**Category**: Headings  
**Auto-fix**: ✓ Available

## Rule Description

This rule ensures closed ATX-style headings have no spaces inside the closing hashes. Closed ATX headings use trailing hash characters for symmetry, and spaces inside these markers are incorrect.

## Why This Rule Exists

Proper closed heading format is important because:

- Follows the ATX heading specification
- Maintains consistent heading style
- Prevents parsing issues with some markdown processors
- Ensures visual symmetry in closed headings

## Examples

### ❌ Incorrect (violates rule)

```markdown
# Heading with space before closing#


## Another heading ##  
### Heading with spaces ### 
```

### ✅ Correct

```markdown
# Heading with proper closing#


## Another heading##


### Correctly closed heading###


# Open heading is also fine
```

## Configuration

This rule has no configuration options.

## Automatic Fix

This rule supports automatic fixing with `--fix`. The fix will:

- Remove spaces between heading content and closing hashes
- Preserve the heading level and content
- Maintain the closed heading style

### Apply Fix

```bash
# Fix closed heading spacing
mdbook-lint lint --fix docs/

# Preview what would be fixed
mdbook-lint lint --fix --dry-run docs/
```

## When to Disable

Consider disabling this rule if:

- You prefer open ATX headings (without closing hashes)
- Your markdown processor has different requirements for closed headings

### Disable in Config

```toml
# .mdbook-lint.toml
disabled_rules = ["MD020"]
```

### Disable Inline

```markdown
<!-- mdbook-lint-disable MD020 -->
# Heading with space # 
<!-- mdbook-lint-enable MD020 -->
```

## Related Rules

- [MD018](./md018.html) - No space after hash on ATX heading
- [MD019](./md019.html) - Multiple spaces after hash on ATX heading
- [MD021](./md021.html) - Multiple spaces inside hashes on closed ATX heading
- [MD022](./md022.html) - Headings should be surrounded by blank lines
- [MD003](./md003.html) - Heading style

## References

- [Original markdownlint rule](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md020)
- [CommonMark Specification - ATX Headings](https://spec.commonmark.org/0.30/#atx-headings)
