# MD024 - No Duplicate Headings

Multiple headings with the same content.

## Why This Rule Exists

Duplicate headings can confuse readers and break anchor links. Each heading
generates a URL fragment, and duplicates create ambiguous navigation targets.

## Examples

### Incorrect

```markdown
# Guide

## Introduction

Some content.

## Introduction

More content with same heading.
```

### Correct

```markdown
# Guide

## Introduction

Some content.

## Getting Started

Different heading for different section.
```

### Siblings Only Mode

With `siblings_only: true`, duplicates are allowed in different sections:

```markdown
# Chapter 1

## Summary

Chapter 1 summary.

# Chapter 2

## Summary

Chapter 2 summary (allowed - different parent).
```

## Configuration

```toml
[MD024]
siblings_only = false  # Only check sibling headings (default: false)
allow_different_nesting = false  # Allow same text at different levels
```

## When to Disable

- Documents with intentionally repeated section names
- Auto-generated content with structured repetition

## Rule Details

- **Rule ID**: MD024
- **Aliases**: no-duplicate-heading
- **Category**: Content
- **Severity**: Warning
- **Auto-fix**: No

## Related Rules

- [MD025](./md025.md) - Single top-level heading
- [MD051](./md051.md) - Link fragments validation
