# MD023 - Headings Must Start at the Beginning of the Line

**Severity**: Warning  
**Category**: Headings  
**Auto-fix**: ✓ Available

## Rule Description

This rule ensures headings start at the beginning of the line without any leading spaces or tabs. Indented headings are not valid in standard markdown.

## Why This Rule Exists

Headings at line start are important because:

- Indented text with hashes may be interpreted as code or regular text
- Ensures headings are properly recognized by all parsers
- Maintains consistent document structure
- Follows CommonMark specification

## Examples

### ❌ Incorrect (violates rule)

```markdown
  # Indented heading
    ## Another indented heading
    ### Tab-indented heading
```

### ✅ Correct

```markdown
# Heading at line start
## Another proper heading
### Correctly positioned heading
```

## Configuration

This rule has no configuration options.

## Automatic Fix

This rule supports automatic fixing with `--fix`. The fix will:

- Remove all leading whitespace (spaces and tabs) before headings
- Preserve the heading level and content
- Maintain proper heading structure

### Apply Fix

```bash
# Fix indented headings
mdbook-lint lint --fix docs/

# Preview what would be fixed
mdbook-lint lint --fix --dry-run docs/
```

## When to Disable

Consider disabling this rule if:

- You're documenting markdown syntax and showing indented hash examples
- Your content includes code blocks with hash-prefixed comments

### Disable in Config

```toml
# .mdbook-lint.toml
disabled_rules = ["MD023"]
```

### Disable Inline

```markdown
<!-- mdbook-lint-disable MD023 -->
    # This indented text is intentional
<!-- mdbook-lint-enable MD023 -->
```

## Related Rules

- [MD018](./md018.html) - No space after hash on ATX heading
- [MD019](./md019.html) - Multiple spaces after hash on ATX heading
- [MD022](./md022.html) - Headings should be surrounded by blank lines
- [MD025](./md025.html) - Multiple top-level headings in the same document
- [MD026](./md026.html) - Trailing punctuation in heading

## References

- [Original markdownlint rule](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md023)
- [CommonMark Specification - ATX Headings](https://spec.commonmark.org/0.30/#atx-headings)
