# MD028 - No Blanks in Blockquote

Blank line inside blockquote.

## Why This Rule Exists

A blank line inside a blockquote ends the quote in most Markdown parsers.
This can cause unexpected rendering where content intended to be quoted
appears as regular text.

## Examples

### Incorrect

```markdown
> First paragraph of quote.
>

> Second paragraph (this is a new blockquote).
```

### Correct

```markdown
> First paragraph of quote.
>
> Second paragraph (still in same blockquote).
```

### Multiple Paragraphs

```markdown
> This is a long quote that spans
> multiple paragraphs.
>
> The blank line has a `>` marker
> to continue the blockquote.
```

## Configuration

This rule has no configuration options.

## When to Disable

- Documents intentionally using separate blockquotes
- Content where visual separation is desired

## Rule Details

- **Rule ID**: MD028
- **Aliases**: no-blanks-blockquote
- **Category**: Formatting
- **Severity**: Warning
- **Auto-fix**: Yes (adds `>` to blank lines)

## Related Rules

- [MD027](./md027.md) - Multiple spaces after blockquote symbol
