# MD029 - Ordered List Prefix

Ordered list item prefix consistency.

## Why This Rule Exists

Markdown supports different numbering styles for ordered lists. Consistent
style improves readability and makes reordering items easier.

## Styles

### One-Based (All 1s)

```markdown
1. First item
1. Second item
1. Third item
```

Advantage: Easy to reorder without renumbering.

### Sequential (Ordered)

```markdown
1. First item
2. Second item
3. Third item
```

Advantage: Source reflects rendered numbers.

### Zero-Based

```markdown
0. First item
0. Second item
0. Third item
```

## Examples

### Incorrect (Mixed)

```markdown
1. First item
2. Second item
1. Third item
```

### Correct

```markdown
1. First item
2. Second item
3. Third item
```

## Configuration

```toml
[MD029]
style = "one_or_ordered"  # Options: "one", "ordered", "zero", "one_or_ordered"
```

| Value | Description |
|-------|-------------|
| `one` | All items use `1.` |
| `ordered` | Sequential numbering (1, 2, 3...) |
| `zero` | All items use `0.` |
| `one_or_ordered` | Allow either `1.` or sequential |

## When to Disable

- Documents with intentional mixed numbering
- Content using numbers for reference purposes

## Rule Details

- **Rule ID**: MD029
- **Aliases**: ol-prefix
- **Category**: Formatting
- **Severity**: Warning
- **Auto-fix**: Yes

## Related Rules

- [MD004](./md004.md) - Unordered list style
- [MD030](./md030.md) - Spaces after list markers
