# MD038 - Spaces Inside Code Spans

Spaces inside code span markers.

## Why This Rule Exists

Extra spaces inside backticks create inconsistent code formatting. The spaces
become part of the rendered code, which usually isn't intended.

## Examples

### Incorrect

```markdown
Use the ` print() ` function.

Run ` npm install ` to install.
```

### Correct

```markdown
Use the `print()` function.

Run `npm install` to install.
```

### Intentional Spaces

If you need a backtick inside code, use double backticks with spaces:

```markdown
Use `` `backticks` `` for code.
```

## Configuration

This rule has no configuration options.

## When to Disable

- Documents using spaces for specific code formatting
- Content requiring literal spaces in code spans

## Rule Details

- **Rule ID**: MD038
- **Aliases**: no-space-in-code
- **Category**: Code
- **Severity**: Warning
- **Auto-fix**: Yes (removes extra spaces)

## Related Rules

- [MD037](./md037.md) - Spaces inside emphasis
- [MD039](./md039.md) - Spaces inside link text
- [MD040](./md040.md) - Fenced code blocks language
