# MD037 - Spaces Inside Emphasis

Spaces inside emphasis markers.

## Why This Rule Exists

Spaces immediately inside emphasis markers may prevent proper rendering in
some Markdown parsers. The emphasis won't be applied, leaving literal
asterisks or underscores in the output.

## Examples

### Incorrect

```markdown
This is **bold** text.


This is *italic* text.


Here is __also bold__ text.

```

### Correct

```markdown
This is **bold** text.

This is *italic* text.

Here is __also bold__ text.
```

## Configuration

This rule has no configuration options.

## When to Disable

- Generally should not be disabled as spaced emphasis rarely renders correctly

## Rule Details

- **Rule ID**: MD037
- **Aliases**: no-space-in-emphasis
- **Category**: Emphasis
- **Severity**: Warning
- **Auto-fix**: Yes (removes spaces inside markers)

## Related Rules

- [MD038](./md038.md) - Spaces inside code spans
- [MD039](./md039.md) - Spaces inside link text
- [MD049](./md049.md) - Emphasis style
- [MD050](./md050.md) - Strong style
