# MD036 - Emphasis Instead of Heading

Emphasis used instead of a heading.

## Why This Rule Exists

Using bold or italic text on its own line as a pseudo-heading breaks document
structure. Real headings provide proper hierarchy for navigation, accessibility,
and table of contents generation.

## Examples

### Incorrect

```markdown
**Introduction**

This section introduces the topic.

*Getting Started*

Follow these steps to begin.
```

### Correct

```markdown
## Introduction

This section introduces the topic.

## Getting Started

Follow these steps to begin.
```

## Configuration

```toml
[MD036]
punctuation = ".,;:!?。；：！？"  # Punctuation that indicates not a heading
```

Lines ending with punctuation are assumed to be emphasized text, not
pseudo-headings.

## When to Disable

- Documents using emphasis for visual styling
- Content where headings aren't appropriate
- Presentations or slides with different formatting needs

## Rule Details

- **Rule ID**: MD036
- **Aliases**: no-emphasis-as-heading
- **Category**: Emphasis
- **Severity**: Warning
- **Auto-fix**: No

## Why This Matters

Pseudo-headings created with emphasis:

- Don't appear in table of contents
- Break accessibility for screen readers
- Can't be linked to with anchors
- Don't contribute to document outline

## Related Rules

- [MD001](./md001.md) - Heading increment
- [MD003](./md003.md) - Heading style
- [MD022](./md022.md) - Blanks around headings
