# MD039 - Spaces Inside Link Text

Spaces inside link text brackets.

## Why This Rule Exists

Leading or trailing spaces inside link text brackets create inconsistent
formatting and may render with unwanted whitespace in the clickable text.

## Examples

### Incorrect

```markdown
[ Click here ](https://example.com)

[  Documentation  ](./docs.md)

[ Link ](url)
```

### Correct

```markdown
[Click here](https://example.com)

[Documentation](./docs.md)

[Link](url)
```

## Configuration

This rule has no configuration options.

## When to Disable

- Generally should not be disabled as spaced link text is usually unintentional

## Rule Details

- **Rule ID**: MD039
- **Aliases**: no-space-in-links
- **Category**: Links
- **Severity**: Warning
- **Auto-fix**: Yes (removes extra spaces)

## Related Rules

- [MD037](./md037.md) - Spaces inside emphasis
- [MD038](./md038.md) - Spaces inside code spans
- [MD042](./md042.md) - Empty links
- [MD051](./md051.md) - Link fragments
