# MD035 - Horizontal Rule Style

Horizontal rule style should be consistent.

## Why This Rule Exists

Markdown supports multiple horizontal rule syntaxes. Consistent style throughout
a document improves readability and maintainability.

## Styles

```markdown
---

---


---


---


** *

```

All render as horizontal rules but mixing them is inconsistent.

## Examples

### Incorrect

```markdown
Section one content.

---

Section two content.

---


Section three content.
```

### Correct

```markdown
Section one content.

---

Section two content.

---

Section three content.
```

## Configuration

```toml
[MD035]
style = "---"  # Options: "---", "***", "___", "consistent"
```

| Value | Description |
|-------|-------------|
| `---` | Three dashes |
| `***` | Three asterisks |
| `___` | Three underscores |
| `consistent` | Match first occurrence |

## When to Disable

- Documents with intentional style variation
- Content imported from multiple sources

## Rule Details

- **Rule ID**: MD035
- **Aliases**: hr-style
- **Category**: Formatting
- **Severity**: Warning
- **Auto-fix**: Yes

## Related Rules

- [MD003](./md003.md) - Heading style consistency
- [MD004](./md004.md) - Unordered list style consistency
