# MD032 - Blanks Around Lists

Lists should be surrounded by blank lines.

## Why This Rule Exists

Blank lines around lists ensure proper parsing and improve readability. Without
blank lines, some Markdown parsers may not correctly identify list boundaries.

## Examples

### Incorrect

```markdown
Some introductory text.
- Item one
- Item two
Following paragraph.
```

### Correct

```markdown
Some introductory text.

- Item one
- Item two

Following paragraph.
```

## Configuration

This rule has no configuration options.

## When to Disable

- Documents with compact formatting requirements
- Content with tight list-to-text flow

## Rule Details

- **Rule ID**: MD032
- **Aliases**: blanks-around-lists
- **Category**: Structure
- **Severity**: Warning
- **Auto-fix**: Yes

## Related Rules

- [MD022](./md022.md) - Blanks around headings
- [MD031](./md031.md) - Blanks around fenced code blocks
- [MD058](./md058.md) - Blanks around tables
