# MD030 - Spaces After List Markers

**Severity**: Warning  
**Category**: Lists  
**Auto-fix**: ✓ Available

## Rule Description

This rule ensures consistent spacing after list markers (`*`, `-`, `+`, or numbers for ordered lists). Proper spacing improves readability and ensures correct parsing.

## Why This Rule Exists

Consistent list marker spacing is important because:

- Ensures lists are properly recognized by all parsers
- Maintains uniform formatting across documents
- Improves readability and visual structure
- Some parsers require specific spacing for proper rendering

## Examples

### ❌ Incorrect (violates rule)

```markdown
*No space after asterisk
-No space after dash
+No space after plus
1.No space after number

*   Too many spaces
-    Excessive spacing
1.    Too much space in ordered list
```

### ✅ Correct

```markdown
* Single space after asterisk
- Single space after dash
+ Single space after plus
1. Single space after number

* Consistent spacing
  * Nested items also follow rules
    * Multi-level nesting works
```

## Configuration

```toml
[MD030]
ul_single = 1    # Spaces after single-line unordered list marker (default: 1)
ul_multi = 1     # Spaces after multi-line unordered list marker (default: 1)
ol_single = 1    # Spaces after single-line ordered list marker (default: 1)
ol_multi = 1     # Spaces after multi-line ordered list marker (default: 1)
```

## Automatic Fix

This rule supports automatic fixing with `--fix`. The fix will:

- Adjust spacing after list markers to match configuration
- Handle both ordered and unordered lists
- Preserve list content and nesting
- Maintain proper indentation for nested lists

### Apply Fix

```bash
# Fix list marker spacing
mdbook-lint lint --fix docs/

# Preview what would be fixed
mdbook-lint lint --fix --dry-run docs/
```

## When to Disable

Consider disabling this rule if:

- Your markdown processor has different spacing requirements
- You're working with generated content with specific formatting
- Your style guide requires different spacing patterns

### Disable in Config

```toml
# .mdbook-lint.toml
disabled_rules = ["MD030"]
```

### Disable Inline

```markdown
<!-- mdbook-lint-disable MD030 -->
*No space after marker allowed here
<!-- mdbook-lint-enable MD030 -->
```

## Related Rules

- [MD004](./md004.html) - Unordered list style
- [MD005](./md005.html) - Consistent list indentation
- [MD006](./md006.html) - Consider starting lists at the beginning of the line
- [MD007](./md007.html) - Unordered list indentation
- [MD029](./md029.html) - Ordered list item prefix
- [MD032](./md032.html) - Lists surrounded by blank lines

## References

- [Original markdownlint rule](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md030)
- [CommonMark Specification - Lists](https://spec.commonmark.org/0.30/#lists)
