# MD044 - Proper Names Capitalization

Proper names should have correct capitalization.

## Why This Rule Exists

Brand names, product names, and technical terms often have specific
capitalization. Consistent capitalization improves professionalism and
readability.

## Examples

### Configuration

```toml
[MD044]
names = ["JavaScript", "GitHub", "macOS", "iOS"]
code_blocks = false  # Don't check inside code blocks
```

### Incorrect

```markdown
Install the package using Github.

This works on MacOS and IOS devices.

Learn javascript programming.
```

### Correct

```markdown
Install the package using GitHub.

This works on macOS and iOS devices.

Learn JavaScript programming.
```

## Configuration 2


```toml
[MD044]
names = []            # List of proper names with correct capitalization
code_blocks = false   # Check inside code blocks (default: false)
html_elements = false # Check inside HTML elements (default: false)
```

### Common Names

```toml
[MD044]
names = [
  "JavaScript", "TypeScript", "Node.js", "npm",
  "GitHub", "GitLab", "Bitbucket",
  "macOS", "iOS", "iPadOS", "watchOS", "tvOS",
  "MySQL", "PostgreSQL", "MongoDB", "SQLite",
  "Rust", "Python", "Ruby", "Kotlin"
]
```

## When to Disable

- Documents where capitalization varies intentionally
- Code-heavy content where names appear in identifiers
- Historical documents preserving original text

## Rule Details

- **Rule ID**: MD044
- **Aliases**: proper-names
- **Category**: Style
- **Severity**: Warning
- **Auto-fix**: No

## Notes

The rule is smart about context:

- Ignores text inside code blocks (configurable)
- Ignores text inside inline code spans
- Ignores URLs and link destinations

## Related Rules

- [MD038](./md038.md) - Spaces inside code spans
