# MD049 - Emphasis Style

Emphasis style should be consistent.

## Why This Rule Exists

Markdown supports two emphasis markers: asterisks and underscores. Consistent
style improves readability and maintainability.

## Styles

### Asterisks

```markdown
This is *italic* text.
```

### Underscores

```markdown
This is _italic_ text.
```

## Examples

### Incorrect (Mixed)

```markdown
This is *italic* and this is _also italic_.

Use *consistent* formatting _throughout_ the document.
```

### Correct

```markdown
This is *italic* and this is *also italic*.

Use *consistent* formatting *throughout* the document.
```

## Configuration

```toml
[MD049]
style = "asterisk"  # Options: "asterisk", "underscore", "consistent"
```

| Value | Description |
|-------|-------------|
| `asterisk` | Use `*text*` |
| `underscore` | Use `_text_` |
| `consistent` | Match first occurrence |

## When to Disable

- Documents with intentional style variation
- Content imported from multiple sources

## Rule Details

- **Rule ID**: MD049
- **Aliases**: emphasis-style
- **Category**: Formatting
- **Severity**: Warning
- **Auto-fix**: Yes

## Note on Underscores

Underscores inside words are not treated as emphasis:

```markdown
some_variable_name  <!-- Not italic, just text -->
```

## Related Rules

- [MD037](./md037.md) - Spaces inside emphasis
- [MD050](./md050.md) - Strong style
