# MD047 - Files Should End with a Single Newline Character

**Severity**: Warning  
**Category**: Whitespace  
**Auto-fix**: ✓ Available

## Rule Description

This rule ensures files end with exactly one newline character. This is a POSIX standard and helps with version control systems.

## Why This Rule Exists

Files ending with a newline are important because:
- POSIX standard requires text files to end with a newline
- Git and other VCS show "No newline at end of file" warnings
- Prevents issues when concatenating files
- Ensures consistent file formatting
- Some tools expect the trailing newline

## Examples

### ❌ Incorrect (violates rule)

```markdown
# Document

Last line without newline```

Or with multiple newlines:

```markdown
# Document

Last line with multiple newlines


```

### ✅ Correct

```markdown
# Document

Last line with single newline
```

## Configuration

This rule has no configuration options.

## Automatic Fix

This rule supports automatic fixing with `--fix`. The fix will:
- Add a newline if the file doesn't end with one
- Remove extra newlines if there are multiple
- Ensure exactly one newline at the end of the file

### Apply Fix

```bash
# Fix file endings
mdbook-lint lint --fix docs/

# Preview what would be fixed
mdbook-lint lint --fix --dry-run docs/
```

## When to Disable

Consider disabling this rule if:
- You're working with files that intentionally lack final newlines
- Your toolchain doesn't support files with trailing newlines
- You're dealing with generated content that doesn't include newlines

### Disable in Config

```toml
# .mdbook-lint.toml
disabled_rules = ["MD047"]
```

### Disable for Specific Files

Since this is a file-level rule, you can exclude specific files:

```toml
# .mdbook-lint.toml
[[overrides]]
path = "no-newline.md"
disabled_rules = ["MD047"]
```

## Related Rules

- [MD009](./md009.html) - No trailing spaces
- [MD010](./md010.html) - Hard tabs
- [MD012](./md012.html) - Multiple consecutive blank lines

## References

- [Original markdownlint rule](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md047)
- [POSIX definition of text file](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap03.html#tag_03_206)