# MD042 - No Empty Links

No empty links allowed.

## Why This Rule Exists

Empty links with no URL serve no purpose and indicate incomplete content
or a mistake during editing. They create broken user experiences when clicked.

## Examples

### Incorrect

```markdown
Click [here]() for more information.

See the [documentation]().

[Empty link]()
```

### Correct

```markdown
Click [here](https://example.com) for more information.

See the [documentation](./docs.md).

[Valid link](https://example.com)
```

### Placeholder Links

If you need placeholders during drafting, use comments:

```markdown
<!-- TODO: Add link -->
Click [here](#) for more information.
```

## Configuration

This rule has no configuration options.

## When to Disable

- Draft documents with intentional placeholders
- Templates where links are filled programmatically

## Rule Details

- **Rule ID**: MD042
- **Aliases**: no-empty-links
- **Category**: Links
- **Severity**: Error
- **Auto-fix**: No

## Related Rules

- [MD011](./md011.md) - Reversed link syntax
- [MD039](./md039.md) - Spaces inside link text
- [MD051](./md051.md) - Link fragments
- [MD052](./md052.md) - Reference links and images
