# MD056 - Table Column Count

Table column count should be consistent.

## Why This Rule Exists

All rows in a table should have the same number of columns. Mismatched column
counts cause rendering issues and indicate data entry errors.

## Examples

### Incorrect

```markdown
| Header 1 | Header 2 | Header 3 |
|----------|----------|----------|
| Cell 1   | Cell 2   |
| Cell 1   | Cell 2   | Cell 3   | Cell 4 |
```

Row 2 has too few columns, row 3 has too many.

### Correct

```markdown
| Header 1 | Header 2 | Header 3 |
|----------|----------|----------|
| Cell 1   | Cell 2   | Cell 3   |
| Cell 4   | Cell 5   | Cell 6   |
```

### Empty Cells

```markdown
| Header 1 | Header 2 | Header 3 |
|----------|----------|----------|
| Cell 1   |          | Cell 3   |
| Cell 4   | Cell 5   |          |
```

## Configuration

This rule has no configuration options.

## When to Disable

- Tables intentionally using colspan-like behavior
- Content from sources with non-standard table formats

## Rule Details

- **Rule ID**: MD056
- **Aliases**: table-column-count
- **Category**: Structure
- **Severity**: Error
- **Auto-fix**: Yes (adds empty cells)

## Related Rules

- [MD055](./md055.md) - Table pipe style
- [MD058](./md058.md) - Blanks around tables
