# MD058 - Blanks Around Tables

Tables should be surrounded by blank lines.

## Why This Rule Exists

Blank lines around tables ensure proper parsing and improve readability.
Some Markdown parsers require blank lines to correctly identify table
boundaries.

## Examples

### Incorrect

```markdown
Some text here.
| Header 1 | Header 2 |
|----------|----------|
| Cell 1   | Cell 2   |
More text here.
```

### Correct

```markdown
Some text here.

| Header 1 | Header 2 |
|----------|----------|
| Cell 1   | Cell 2   |

More text here.
```

## Configuration

This rule has no configuration options.

## When to Disable

- Documents with compact formatting requirements
- Content where tables flow tightly with surrounding text

## Rule Details

- **Rule ID**: MD058
- **Aliases**: blanks-around-tables
- **Category**: Formatting
- **Severity**: Warning
- **Auto-fix**: Yes

## Related Rules

- [MD022](./md022.md) - Blanks around headings
- [MD031](./md031.md) - Blanks around fenced code blocks
- [MD032](./md032.md) - Blanks around lists
- [MD055](./md055.md) - Table pipe style
- [MD056](./md056.md) - Table column count
