# MD053 - Link and Image Reference Definitions

Link and image reference definitions should be needed.

## Why This Rule Exists

Unused reference definitions clutter documents and may indicate dead links
or incomplete edits. Keeping only used definitions improves maintainability.

## Examples

### Incorrect

```markdown
Check out the [documentation](https://docs.example.com).

[unused]: https://example.com
[also-unused]: https://other.com
```

The definitions are never used.

### Correct

```markdown
Check out the [documentation][docs].

[docs]: https://docs.example.com
```

Or remove unused definitions:

```markdown
Check out the [documentation](https://docs.example.com).
```

## Configuration

```toml
[MD053]
ignored_definitions = []  # Definitions to ignore (e.g., for includes)
```

### Ignoring Definitions

```toml
[MD053]
ignored_definitions = ["//", "TODO"]
```

## When to Disable

- Documents with definitions used in included content
- Templates with conditional reference usage
- Files serving as definition libraries

## Rule Details

- **Rule ID**: MD053
- **Aliases**: link-image-reference-definitions
- **Category**: Links
- **Severity**: Warning
- **Auto-fix**: No

## Related Rules

- [MD052](./md052.md) - Reference links must be defined
- [MD054](./md054.md) - Link and image style
