# MD059 - Descriptive Link Text

Link text should be descriptive.

## Why This Rule Exists

Generic link text like "click here" or "this link" provides no context about
the destination. Descriptive text improves accessibility and helps users
understand where links lead.

## Examples

### Incorrect

```markdown
For more information, [click here](https://docs.example.com).

See [this link](./guide.md) for details.

[Here](https://api.example.com) is the API documentation.

Read more [here](./faq.md).
```

### Correct

```markdown
For more information, see the [complete documentation](https://docs.example.com).

See the [installation guide](./guide.md) for details.

Read the [API documentation](https://api.example.com).

Read the [frequently asked questions](./faq.md).
```

## Configuration

This rule has no configuration options.

## When to Disable

- UI documentation where "click here" matches actual button text
- Content with intentionally brief link descriptions

## Rule Details

- **Rule ID**: MD059
- **Aliases**: descriptive-link-text
- **Category**: Accessibility
- **Severity**: Warning
- **Auto-fix**: No

## Common Non-Descriptive Phrases

The rule flags these common patterns:

- "click here"
- "here"
- "this link"
- "this"
- "link"
- "read more"

## Why Descriptive Text Matters

1. **Screen Readers**: Users often navigate by links; "click here" provides no context
2. **Scanning**: Users scan pages looking for relevant links
3. **SEO**: Search engines use link text to understand content relationships
4. **Mobile**: Touch targets benefit from clear labels

## Accessibility Standards

This rule helps comply with:

- WCAG 2.1 Success Criterion 2.4.4 (Link Purpose in Context)
- WCAG 2.1 Success Criterion 2.4.9 (Link Purpose Link Only)

## Related Rules

- [MD042](./md042.md) - No empty links
- [MD045](./md045.md) - Images should have alt text
