# MD050 - Strong Style

Strong emphasis style should be consistent.

## Why This Rule Exists

Markdown supports two strong emphasis markers: double asterisks and double
underscores. Consistent style improves readability.

## Styles

### Asterisks (Common)

```markdown
This is **bold** text.
```

### Underscores

```markdown
This is __bold__ text.
```

## Examples

### Incorrect (Mixed)

```markdown
This is **bold** and this is __also bold__.
```

### Correct

```markdown
This is **bold** and this is **also bold**.
```

## Configuration

```toml
[MD050]
style = "asterisk"  # Options: "asterisk", "underscore", "consistent"
```

| Value | Description |
|-------|-------------|
| `asterisk` | Use `**text**` |
| `underscore` | Use `**text**` |

| `consistent` | Match first occurrence |

## When to Disable

- Documents with intentional style variation
- Content imported from multiple sources

## Rule Details

- **Rule ID**: MD050
- **Aliases**: strong-style
- **Category**: Formatting
- **Severity**: Warning
- **Auto-fix**: Yes

## Related Rules

- [MD037](./md037.md) - Spaces inside emphasis
- [MD049](./md049.md) - Emphasis style
