# MD054 - Link and Image Style

Link and image style should be consistent.

## Why This Rule Exists

Markdown supports inline and reference-style links. Consistent style throughout
a document improves readability and maintainability.

## Styles

### Inline

```markdown
[Link text](https://example.com)
![Alt text](image.png)
```

### Reference (Full)

```markdown
[Link text][ref]
![Alt text][img]

[ref]: https://example.com
[img]: image.png
```

### Reference (Collapsed)

```markdown
[Example][]

[Example]: https://example.com
```

### Reference (Shortcut)

```markdown
[Example]

[Example]: https://example.com
```

## Examples

### Incorrect (Mixed)

```markdown
See [inline link](https://example.com) and [reference link][ref].

[ref]: https://other.com
```

### Correct

```markdown
See [inline link](https://example.com) and [other link](https://other.com).
```

## Configuration

```toml
[MD054]
autolink = true           # Allow autolinks <https://example.com>
inline = true             # Allow inline style
full = true               # Allow full reference style
collapsed = true          # Allow collapsed reference style
shortcut = true           # Allow shortcut reference style
url_inline = true         # Allow inline for URLs only
```

## When to Disable

- Documents with intentional style mixing
- Content where different styles serve different purposes

## Rule Details

- **Rule ID**: MD054
- **Aliases**: link-image-style
- **Category**: Links
- **Severity**: Warning
- **Auto-fix**: No

## Related Rules

- [MD052](./md052.md) - Reference links must be defined
- [MD053](./md053.md) - Unused reference definitions
- [MD059](./md059.md) - Descriptive link text
